kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/assertions/clues.kt

```diff
@@ -20,7 +37,7 @@ inline fun <R> withClue(clue: Any, thunk: () -> R): R {
  */
 inline fun <T : Any, R> T.asClue(block: (T) -> R): R {
    try {
-      errorCollector.pushClue(this)
+      errorCollector.pushClue { this.toString() }
```

Comment:
could probably just delegate this entire method to the other withClue

inline fun <T : Any, R> T.asClue(block: (T) -> R): R = withClue(lazy { this.toString() }, block)